omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java

```diff
@@ -55,10 +64,57 @@ protected long getLastModified(HttpServletRequest servletRequest) {
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-        if (SpaModuleUtils.isRemoteAssetsEnabled()) {
-            handleRemoteAssets(request, response);
+        String requestURI = request.getRequestURI();
+        if (requestURI != null && requestURI.contains("frontend/config")) {
+            // get the config.json file
+            File configFile = getJsonConfigFile();
+            if (configFile.exists()) {
+                response.setContentType("application/json");
+                response.setHeader("Content-Disposition", "attachment; filename=\"" + configFile.getAbsolutePath() + "\"");
+
+                FileInputStream input = new FileInputStream(configFile);
+                byte[] buffer = new byte[1024];
+                int bytesRead;
+                while ((bytesRead = input.read(buffer)) != -1) {
+                    response.getOutputStream().write(buffer, 0, bytesRead);
+                }
+                input.close();
+            } else {
+                response.sendError(HttpServletResponse.SC_NOT_FOUND);
+            }
         } else {
-            handleLocalAssets(request, response);
+            if (SpaModuleUtils.isRemoteAssetsEnabled()) {
+                handleRemoteAssets(request, response);
+            } else {
+                handleLocalAssets(request, response);
+            }
+        }
+    }
+    @Override
+    protected void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        String requestURI = request.getRequestURI();
+        if (requestURI.contains("frontend/config")) {
+
+            BufferedReader reader = request.getReader();
+            StringBuilder stringBuilder = new StringBuilder();
+            String line;
+
+            while ((line = reader.readLine()) != null) {
+                stringBuilder.append(line);
+            }
+
+            String requestBody = stringBuilder.toString();
+
+            File jsonConfigFile = getJsonConfigFile();
+            FileWriter writer = new FileWriter(jsonConfigFile);
+            writer.write(requestBody);
+            writer.close();
+
+            if (jsonConfigFile.exists()) {
+                response.setStatus(HttpServletResponse.SC_OK);
+            } else {
+                response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
```

Comment:
It would be a good idea to log something and return some kind of message about what failed...